core/util_test.go

```diff
@@ -345,33 +432,51 @@ func (d *DrandTestScenario) RunDKG() *key.Group {
 		required.NoError(e)
 	}
 
-	d.t.Logf("[RunDKG] Leader %s FINISHED", leaderNode.addr)
+	t.Logf("[RunDKG] Leader %s FINISHED", leaderNode.addr)
 
 	// we check that we can fetch the group using control functionalities on the leaderNode node
 	groupProto, err := controlClient.GroupFile(d.beaconID)
-	require.NoError(d.t, err)
-	d.t.Logf("[-------] Leader %s FINISHED", leaderNode.addr)
+	require.NoError(t, err)
+	t.Logf("[-------] Leader %s FINISHED", leaderNode.addr)
 	group, err := key.GroupFromProto(groupProto)
-	require.NoError(d.t, err)
+	require.NoError(t, err)
 
 	// we check all nodes are included in the group
 	for _, node := range d.nodes {
-		require.NotNil(d.t, group.Find(node.drand.priv.Public))
+		require.NotNil(t, group.Find(node.drand.priv.Public))
 	}
 
 	// we check the group has the right threshold
-	require.Len(d.t, group.PublicKey.Coefficients, d.thr)
-	require.Equal(d.t, d.thr, group.Threshold)
-	require.NoError(d.t, key.Save(d.groupPath, group, false))
+	require.Len(t, group.PublicKey.Coefficients, d.thr)
+	require.Equal(t, d.thr, group.Threshold)
+	require.NoError(t, key.Save(d.groupPath, group, false))
 
 	d.group = group
-	d.t.Log("[RunDKG] READY!")
+	t.Log("[RunDKG] READY!")
 	return group
 }
 
-func (d *DrandTestScenario) Cleanup() {
-	_ = os.RemoveAll(d.dir)
-	_ = os.RemoveAll(d.newDir)
+func (d *DrandTestScenario) Cleanup(t *testing.T) {
+	d.CloseAllDaemons(t)
+
+	if len(d.logFiles) != 0 {
+		closeLogFiles(t, d.logFiles, d.dir)
+		return
+	}
+
+	// Do not perform a cleanup of temp dirs if we write logs to files
+	// when using DRAND_TEST_FILE_LOGS=true env var setting
```

Comment:
This doesn't seem compatible with the usage of `t.TempDir()` since it got automagically cleanup by the `t.Cleanup` at the end of the tests.

Maybe we should have `DRAND_TEST_FILE_LOGS=/path/to/desired/log/folder` instead?

Reply:
That's a great idea!
I'll switch to that once I fix the flackyness which caused these tests to fail.